SQL_Assignment_2.sql

```diff
@@ -0,0 +1,165 @@
+
+SELECT * FROM EMPLOYEES
+WHERE LASTNAME LIKE 'B%' OR LASTNAME LIKE 'R%' OR LASTNAME LIKE 'S'
+
+--Retrieve all employees whose first name begins with any letter from 'A' through 'M'.
+
+SELECT * FROM EMPLOYEES
+WHERE LEFT(FIRSTNAME,1) BETWEEN 'A' AND 'M';
+
+--List employees whose last name doesn't start with a vowel (A, E, I, O, U).
+
+SELECT * FROM EMPLOYEES
+WHERE LEFT(LASTNAME,1) NOT IN ('A','E','I','O','U');
+
+--Identify employees earning more than $80,000 annually. 
+
+SELECT *
+FROM EMPLOYEES 
+WHERE SALARY>80000
+
+--Find employees who joined the company before 2020. 
+
+SELECT *
+FROM EMPLOYEES 
+WHERE YEAR(HIREDATE)<2020
+
+-- List all employees not named 'John' (first name).
+
+SELECT *
+FROM EMPLOYEES 
+WHERE FIRSTNAME !='JOHN'
+
+
+--Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+
+SELECT *
+FROM EMPLOYEES
+WHERE DEPARTMENT='MARKETING' AND (SALARY <=60000 OR HIREDATE > '2019-06-30');
+
+--Find employees whose first name contains the letters 'an' anywhere and ends with 'e'.
+
+SELECT *
+FROM EMPLOYEES
+WHERE  FIRSTNAME LIKE '%AN%' AND FIRSTNAME LIKE '%E';    
+
+CREATE TABLE Sales (
+    SaleID INT PRIMARY KEY,
+    ProductID INT,
+    CustomerID INT,
+    SaleDate DATE,
+    Quantity INT,
+    UnitPrice DECIMAL(10, 2)
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(100),
+    Category VARCHAR(50)
+);
+
+CREATE TABLE Customers (
+    CustomerID INT PRIMARY KEY,
+    CustomerName VARCHAR(100),
+    City VARCHAR(50),
+    State VARCHAR(2)
+);
+
+
+--Calculate the total sales amount for each product. Display the ProductID and total sales amount.
+
+SELECT PRODUCTID , SUM(QUANTITY*UNITPRICE) AS TOTALSALESAMOUNT
+FROM SALES
+GROUP BY PRODUCTID
+
+--Find the average quantity sold per sale for each product category.
+
+SELECT p.Category, AVG(s.Quantity) AS AverageQuantity
+FROM Sales s
+JOIN Products p ON s.ProductID = p.ProductID
+GROUP BY p.Category;
+
+--List the top 5 customers by their total purchase amount. Include the CustomerID and total purchase amount.
+
+SELECT TOP 5 CustomerID, SUM(Quantity * UnitPrice) AS TotalPurchaseAmount
+FROM Sales
+GROUP BY CustomerID
+ORDER BY TotalPurchaseAmount DESC;
+
+--Determine the number of sales made each month in the year 2023. Display the month and the count of sales.
+
+SELECT MONTH(SALEDATE) AS SALES_IN_MONTH, COUNT(SALEID) AS COUNT_OF_SALES
+FROM SALES
+WHERE YEAR(SALEDATE)='2023'
+GROUP BY MONTH(SALEDATE)
+
+--Calculate the total revenue for each state, but only for states with more than $10,000 in total sales.
+SELECT C.STATE AS STATE,SUM(S.QUANTITY*S.UNITPRICE)  
+FROM SALES S  
+JOIN CUSTOMERS C ON C.CUSTOMERID=S.CUSTOMERID
+GROUP BY C.STATE
+HAVING SUM(S.QUANTITY*S.UNITPRICE) >10000
+;
+
+--Find products that have been sold more than 100 times in total. Display the ProductID, ProductName, and the total quantity sold.
+SELECT P.PRODUCTID, P.PRODUCTNAME FROM
```

Comment:
sold more than 100 times

Reply:
My mistake, my db didnt had 100 product in the table so i tried with 5